### PR TITLE
Home work 28

### PR DESCRIPTION
**Pull request by UnitOmega13**

### CODE REVIEW

File: src/src/main/java/controller/NewUserServlet.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(name = "NewUserServlet", value = "/admin/users/add")
+public class NewUserServlet extends HttpServlet {
+
+    private static final UserService userService = UserServiceFactory.getInstance();
+    private static Long id = IdGenerator.generateId();
```
Comment (Sofasmile) [inline src/src/main/java/controller/NewUserServlet.java]: better move in the method

File: src/src/main/java/controller/ProductRegistrationServlet.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/new_product")
+public class ProductRegistrationServlet extends HttpServlet {
+
+    private static final ProductService productService = ProductServiceFactory.getInstance();
+    private static Long id = IdGenerator.generateId();
```
Comment (Sofasmile) [inline src/src/main/java/controller/ProductRegistrationServlet.java]: the same

File: src/src/main/java/controller/SignUpServlet.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/registration")
+public class SignUpServlet extends HttpServlet {
+
+    private static final UserService userService = UserServiceFactory.getInstance();
+    private static Long id = IdGenerator.generateId();
```
Comment (Sofasmile) [inline src/src/main/java/controller/SignUpServlet.java]: :)

File: src/src/main/java/dao/UsersDAO.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package dao;
+
+import model.User;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface UsersDAO{
```
Comment (Sofasmile) [inline src/src/main/java/dao/UsersDAO.java]:
```suggestion
public interface UsersDAO {
```

File: src/src/main/java/service/Impl/MailServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+import java.util.Properties;
+
+public class MailServiceImpl implements MailService {
+    private static final BasketService basketService = BasketServiceFactory.getInstance();
+    private static final String userNmae = "testu0513@gmail.com";
+    private static final String password = "490401Neko";
+
+    private Properties getProperties() {
+        Properties prop = new Properties();
```
Comment (Sofasmile) [inline src/src/main/java/service/Impl/MailServiceImpl.java]:
```suggestion
        Properties property = new Properties();
```

File: src/src/main/java/model/Basket.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Arrays;
+
+public class Basket {
+    private User user;
+    private List<Optional<Product>> products = new ArrayList<Optional<Product>>();
```
Comment (Sofasmile) [inline src/src/main/java/model/Basket.java]: do we need optional here?

File: src/src/main/java/service/Impl/MailServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+            message.setFrom(new InternetAddress(userNmae));
+            message.setRecipients(
+                    Message.RecipientType.TO,
+                    InternetAddress.parse(user.getEmail())
+            );
+            message.setSubject("Code for confirmation order");
+            message.setText(user.getCode());
+            Transport.send(message);
+        } catch (MessagingException e) {
+            e.printStackTrace();
```
Comment (Sofasmile) [inline src/src/main/java/service/Impl/MailServiceImpl.java]: ?  '_'

File: src/src/main/java/service/Impl/MailServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+            message.setRecipients(
+                    Message.RecipientType.TO,
+                    InternetAddress.parse(user.getEmail())
+            );
+            message.setSubject("Your order " + order.getId());
+            message.setText(basketService.getUserBasket(user).get().toString() +
+                    "\n" + order.toString());
+            Transport.send(message);
+        } catch (MessagingException e) {
+            e.printStackTrace();
```
Comment (Sofasmile) [inline src/src/main/java/service/Impl/MailServiceImpl.java]: ?

File: src/src/main/java/utils/IdGenerator.java [hunk 9]
```diff
@@ -0,0 +1,7 @@
+package utils;
+
+public class IdGenerator {
+    public static long generateId(){
+        return (long) (Math.random() * 1000000);
```
Comment (Sofasmile) [inline src/src/main/java/utils/IdGenerator.java]: we need increment id all time! from 0 to ....

Comment (UnitOmega13) [reply src/src/main/java/utils/IdGenerator.java]: fixed

**Pull request closed without merging**